src/Microsoft.Identity.Client/Platforms/netcore/OsBrowser/TcpInterceptor.cs

```diff
@@ -0,0 +1,135 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Net;
+using System.Net.Sockets;
+using System.Text;
+using System.Text.RegularExpressions;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Identity.Client.Platforms.netcore.OsBrowser
+{
+    /// <summary>
+    /// This object is responsible for listening to a single TCP request, on localhost:port, 
+    /// extracting the uri, parsing 
+    /// </summary>
+    /// <remarks>
+    /// The underlying TCP listener might capture multiple requests, but only the first one is handled.
+    /// </remarks>
+    internal class TcpInterceptor : ITcpInterceptor
```

Comment:
Most of the code in here isn't really testable directly, however we use this logic for our Selenium tests so this is getting a bit of use.